### PR TITLE
Feature/sonic boom

### PR DESCRIPTION
**Pull request by jaswsinc**
Important files to review as a part of the work in this PR:

- See [cache.txt](https://github.com/websharks/zencache-pro/blob/a5e01ef319daa3aeaf2fcb4ea235ff149259e9c1/src/includes/templates/htaccess/cache.txt), which serves cache files via `.htaccess` and some jujumagumbo.

_Noting that this is not yet MS compatible._
_Noting that all of our `.htaccess` tweaks require Apache 2.2+_

## TODO

- [ ] Update [this class member](https://github.com/websharks/zencache-pro/blob/000000-dev/src/includes/closures/Shared/CachePathUtils.php#L16) so that it considers the current mode of operation.
- [ ] Figure out how to deal with stored headers when serving cache files via `.htaccess`.
- [ ] Figure out how to deal with GZIP compression coupled with the possible use of `mod_asis`.
- [ ] Add a mode that will get this working on Ningx also; and perhaps resolve https://github.com/websharks/zencache/issues/435.

### CODE REVIEW

File: src/includes/classes/Plugin.php [hunk 1]
```diff
@@ -338,7 +348,10 @@ public function setup()
             'auto_cache_delay'       => '500', // In milliseconds.
             'auto_cache_sitemap_url' => 'sitemap.xml', // Relative to `site_url()`.
             'auto_cache_other_urls'  => '', // A line-delimited list of any other URLs.
-            'auto_cache_user_agent'  => 'WordPress',
+            'auto_cache_permalinks'  => '1', // `0|1`; auto-cache known Post/Page permalinks?
```
Comment (raamdev) [inline src/includes/classes/Plugin.php]: I'm curious what these additional Auto-Cache options are for. Is there an associated GitHub issue?

Comment (jaswsinc) [inline src/includes/classes/Plugin.php]: I haven't figured this out yet, sorry.

My thinking is that I will try pull together a list of known permalinks and archives based on what is in the DB. That way the ACE will be capable of pre-caching a site for the most part, even when it lacks an XML sitemap.

This is a part of trying to accomplish a mode whereby ZenCache options could be preconfigured in such a way to create a sonic boom with very little effort on the part of a site owner. Those two options may or may not become a part of the UI directly, but having them in there might make it possible for different modes to be preconfigured in a future release.

Comment (jaswsinc) [inline src/includes/classes/Plugin.php]: ha.. I am actually using the term sonic boom when I talk about ZenCache now. That's funny.

Comment (raamdev) [inline src/includes/classes/Plugin.php]: > ha.. I am actually using the term sonic boom when I talk about ZenCache now.

We should totally work that into the UI somehow... "**Enable (safest)** | **Enable (fastest)** | **Enable (sonic boom!)**"

**Comment by jaswsinc** [comment]
Adding TODO items above. Still needs research and some thought.

**Comment by raamdev** [comment]
@jaswsinc Just a heads up: You may want to merge `000000-dev` into this branch sooner rather than later. It looks like there are already some merge conflicts you'll need to resolve. :-)

**Comment by raamdev** [comment]
Also noting that merging `000000-dev` will cause the `assets/` directory to now be managed under LFS.

**Comment by jaswsinc** [comment]
Copy that. I'm going to leave this here for now, but I suspect I will need to start over on this effort to create a sonic boom. My feeling is that [these changes for 5.4](https://github.com/websharks/zencache/issues/635) should come first, and since that work is going to make it impossible to merge this branch in after, I'm thinking I should work on 5.4 enhancements first and come back to this later and try again.

**Comment by raamdev** [comment]
@jaswsinc I'm wondering if you'd be able to redo this PR now that the codebase has been updated for PHP 5.4. Doing so would give me a big head-start on everything that's going to go into a new **Apache Optimizations** panel, including the ability to use `mod_rewrite` to load cached files (see https://github.com/websharks/comet-cache/issues/783).

**Comment by raamdev** [comment]
@jaswsinc Just noting that @renzms is currently working on several issues related to the **Apache Optimizations** panel (including adding the panel itself). You may want to hold off on redoing this PR until after that work:

https://github.com/websharks/comet-cache/issues/765
https://github.com/websharks/comet-cache/issues/764
https://github.com/websharks/comet-cache/issues/554

**Comment by jaswsinc** [comment]
Copy that. Thank you.

**Comment by jaswsinc** [comment]
I'm closing this, but leaving the branch intact, for now; i.e., for future reference.

**Pull request closed without merging**